CUPS checks for "-fPIE", but uses "-fPIE -pie". Make it check for what it actually uses.
--- config-scripts/cups-compiler.m4.~1~	2011-06-11 02:59:36.000000000 +0400
+++ config-scripts/cups-compiler.m4	2016-05-13 22:01:03.821851358 +0300
@@ -152,7 +152,16 @@
 	AC_MSG_CHECKING(if GCC supports -fPIE)
 	OLDCFLAGS="$CFLAGS"
 	CFLAGS="$CFLAGS -fPIE"
-	AC_TRY_COMPILE(,,
+	case "$CC" in
+		*clang)
+			CFLAGS="$CFLAGS -fPIE -Wl,-pie"
+		;;
+		*)
+			CFLAGS="$CFLAGS -fPIE -pie"
+		;;
+	esac
+
+	AC_TRY_LINK(,,
 		[case "$CC" in
 			*clang)
 				PIEFLAGS="-fPIE -Wl,-pie"
